lib/src/native/realm_core.dart

```diff
@@ -1811,59 +1888,64 @@ class ConfigHandle extends HandleBase<realm_config> {
 }
 
 class RealmHandle extends HandleBase<shared_realm> {
+  int _counter = 0;
+
+  final Map<int, WeakReference<RootedHandleBase>> _children = {};
+
   RealmHandle._(Pointer<shared_realm> pointer) : super(pointer, 24);
 
   RealmHandle._unowned(Pointer<shared_realm> pointer) : super.unowned(pointer);
+
+  int addChild(RootedHandleBase child) {
+    final id = _counter++;
+    _children[id] = WeakReference(child);
+    return id;
+  }
+
+  void removeChild(int id) {
+    _children.remove(id);
+  }
+
+  @override
+  void _releaseCore() {
+    final keys = _children.keys.toList();
```

Comment:
Do we need the `toList` here. Can't we do a `.forEach`. or iterate over the `_children` collection